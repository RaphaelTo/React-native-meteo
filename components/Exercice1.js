import React, { Fragment, Component } from 'react'
import { View, Text ,StyleSheet, Animated } from 'react-native'

export default class Exercice1 extends Component {
    
    state= {
        top: new Animated.Value(0),
        bottom: new Animated.Value(0)
    }
    
    componentDidMount() {
        Animated.parallel([
            Animated.timing(this.state.bottom, {
                toValue: 300,
                duration: 500
            }),
            Animated.timing(this.state.top, {
                toValue: 0,
                duration: 500
            }),
        ]).start()
    }
    
    render(){

        const {top, bottom} = this.state
        const styles = StyleSheet.create({
            firstview: {
                flex: 1,
                backgroundColor: '#fff',
                paddingBottom: 0,
                alignItems: 'center',
                justifyContent: 'center',
              },
              secondview: {
                  flex: 1,
                  backgroundColor: 'red',
                  paddingTop: 0,
                  alignItems: 'center',
                  justifyContent: 'center',
                },
        })

        return (
            <Fragment>
                <Animated.View>
                    <View style={styles.firstview}>
                        <Text>Slide 1</Text>
                    </View>
                    <View style={styles.secondview}>
                        <Text>Slide 2</Text>
                    </View>
                </Animated.View>
            </Fragment>
        )
    }
}

