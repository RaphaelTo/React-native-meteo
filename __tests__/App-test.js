/**
 * @format
 */

import 'react-native';
import React from 'react';
import Exercice1 from '../components/Exercice1'
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(<Exercice1 />).toJSON();
  expect(tree).toMatchSnapshot();
});
