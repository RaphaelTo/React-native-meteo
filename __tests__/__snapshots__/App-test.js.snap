// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "flex": 1,
        "justifyContent": "center",
        "paddingBottom": 0,
      }
    }
  >
    <Text>
      Slide 1
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "red",
        "flex": 1,
        "justifyContent": "center",
        "paddingTop": 0,
      }
    }
  >
    <Text>
      Slide 2
    </Text>
  </View>
</View>
`;
